src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -1303,6 +1306,52 @@ class IfClauseSegment(BaseSegment):
     )
 
 
+class WhileExpressionStatement(BaseSegment):
+    """WHILE statement.
+
+    https://docs.microsoft.com/en-us/sql/t-sql/language-elements/while-transact-sql?view=sql-server-ver15
+    """
+
+    type = "while_statement"
+
+    match_grammar = Sequence(
+        "WHILE",
+        Ref("ExpressionSegment"),
+        Indent,
+        Sequence(
+            Ref("StatementSegment"),
+            Ref("DelimiterSegment", optional=True),
+        ),
```

Comment:
We use this in a lot of places. Probably should define a Grammar for it, to save retyping it over and over again.